packages/expo-dev-menu/ios/DevMenuRCTBridge.mm

```diff
@@ -106,25 +119,38 @@ - (void)reloadWithReason:(NSString *)reason {}
 
 @end
 
-@interface DevMenuRCTCxxBridgeDelegate () <RCTCxxBridgeDelegate>
-
-@end
-
-@implementation DevMenuRCTCxxBridgeDelegate
+@implementation DevMenuRCTAppDelegate
 
 - (std::unique_ptr<facebook::react::JSExecutorFactory>)jsExecutorFactoryForBridge:(RCTBridge *)bridge
 {
-#if __has_include(<reacthermes/HermesExecutorFactory.h>)
-  // Disable Hermes debugger to prevent Hermes debugger uses dev-menu
-  // as inspecting target.
-  auto installBindings = facebook::react::RCTJSIExecutorRuntimeInstaller(nullptr);
-  auto *hermesExecutorFactory = new facebook::react::HermesExecutorFactory(installBindings);
-  hermesExecutorFactory->setEnableDebugger(false);
-  std::unique_ptr<facebook::react::JSExecutorFactory> jsExecutorFactory(hermesExecutorFactory);
-  return std::move(jsExecutorFactory);
-#else
-  return nullptr;
-#endif
+  #if RCT_NEW_ARCH_ENABLED
+    self.turboModuleManager = [[RCTTurboModuleManager alloc] initWithBridge:bridge
+                                                                  delegate:(id<RCTTurboModuleManagerDelegate>)self
+                                                                  jsInvoker:bridge.jsCallInvoker];
+    std::unique_ptr<facebook::react::JSExecutorFactory> executorFactory = RCTAppSetupDefaultJsExecutorFactory(bridge, self.turboModuleManager);
+
+    #if __has_include(<reacthermes/HermesExecutorFactory.h>)
+        auto rawExecutorFactory = executorFactory.get();
+        auto hermesExecFactory = dynamic_cast<facebook::react::HermesExecutorFactory*>(rawExecutorFactory);
+        if (hermesExecFactory != nullptr) {
+            hermesExecFactory->setEnableDebugger(false);
+        }
+    #endif
+
+    return executorFactory;
+  #endif
```

Comment:
@Kudo do you think it's fine to disable the debugger like this?

Reply:
this is intentional. hermes debugger does not well support loading different bundles. the latest loading bundle will override the previous one from debugging. we had an issue in dev-client when dev-menu shown, the debugger target will be overridden as dev-menu's bundle. people should reload the app to override the debugging target back to the app.
with the `setEnableDebugger` we can prevent the problem.

Reply:
But do you think this casting from `JSExecutorFactory` to `HermesExecutorFactory` would still work as intended?
I'm trying to keep the `RCTAppSetupDefaultJsExecutorFactory(bridge, self.turboModuleManager)` call

Reply:
thanks for the effort to use the `RCTAppSetupDefaultJsExecutorFactory`. most react-native code requires `-frtti` so it should be okay to do the dynamic_cast here. moreover, you could also try static_cast if that works, given the fact that inside `#if __has_include(<reacthermes/HermesExecutorFactory.h>)` block, the executor should be hermes.